src/main/java/com/esri/samples/localserver/local_server_dynamic_workspace_raster/LocalServerDynamicWorkspaceRasterSample.java

```diff
@@ -0,0 +1,164 @@
+          new ExtensionFilter("Image Files", "*.tif"));
+      fileChooser.setInitialDirectory(new File("./samples-data/raster/"));
+
+      // create button to open file chooser
+      Button addButton = new Button("Choose Raster");
+      addButton.setMaxWidth(Double.MAX_VALUE);
+      addButton.setDisable(true);
+      addButton.setOnAction(e -> {
+       File rasterFile = fileChooser.showOpenDialog(stage);
+       if(rasterFile != null){
+         startLocalService(rasterFile);
+       }
+      });
+
+      // add button to the control panel
+      vBoxControl.getChildren().addAll(description, addButton);
+
+      // create a map and add it to a map view
+      ArcGISMap map = new ArcGISMap(Basemap.createTopographic());
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // start the local server
+      server.startAsync().addDoneListener(() -> {
+        if (server.getStatus() == LocalServerStatus.STARTED) {
+          addButton.setDisable(false);
+        }
+      });
+
+      // add the map view and control panel to stack pane
+      stackPane.getChildren().addAll(mapView, vBoxControl);
+      StackPane.setAlignment(vBoxControl, Pos.TOP_LEFT);
+      StackPane.setMargin(vBoxControl, new Insets(10, 0, 0, 10));
+
+    } catch (Exception e) {
+      // on any error, display the stack trace
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Start a LocalMapService with the Raster File.
+   *
+   * @param rasterFile a raster file
+   */
+  private void startLocalService(File rasterFile) {
+    // start a service from a blank map package
+    String mapServiceURL = "./samples-data/local_server/mpk_blank.mpk";
+    LocalMapService localMapService = new LocalMapService(mapServiceURL);
+
+    // create a raster workspace
+    String fileName = rasterFile.getName();
+    String path = rasterFile.getParent();
+    RasterWorkspace rasterWorkspace = new RasterWorkspace("raster_wkspc", path);
+    RasterSublayerSource source = new RasterSublayerSource(rasterWorkspace.getId(), fileName);
+    ArcGISMapImageSublayer imageSublayer = new ArcGISMapImageSublayer(101, source);
+    List<DynamicWorkspace> dynamicWorkspaces = Collections.singletonList(rasterWorkspace);
+    localMapService.setDynamicWorkspaces(dynamicWorkspaces);
+
+    // start the local map service
+    localMapService.startAsync();
+    localMapService.addStatusChangedListener(event -> {
+      if (event.getNewStatus() == LocalServerStatus.STARTED) {
+        // create a map image layer with the URL of the local map service
+        ArcGISMapImageLayer imageLayer = new ArcGISMapImageLayer(localMapService.getUrl());
+
+        // add the sublayer to the image layer
+        imageLayer.getSublayers().add(imageSublayer);
+        imageLayer.loadAsync();
+
+        // add the image layer to the map as an operational layer
+        mapView.getMap().getOperationalLayers().add(imageLayer);
+        mapView.setViewpoint(new Viewpoint(imageLayer.getFullExtent()));
```

Comment:
This extent is only getting the extent of the `blank.mpk` so it will always be the same no matter what the raster is.
I thought that `imageSublayer.getMapServiceSublayerInfo().getExtent()` should work but for some reason the `Load Status` of the imageSublayer is `Not Loaded`. This make no sense because I can see it in the view so it should be loaded. Feels like a bug to me.